lib/src/lifecycle_module.dart

```diff
@@ -51,14 +51,21 @@ enum LifecycleState {
   unloaded
 }
 
+/// Contains "original" module API for backwards compatibility.
+abstract class LifecycleModule extends _LifecycleModule<LifecycleModule>
+    with SimpleModule {}
+
+/// Only contains lifecycle-related methods and properties.
+abstract class LifecycleModuleV2 extends _LifecycleModule<LifecycleModuleV2> {}
+
 /// Intended to be extended by most base module classes in order to provide a
 /// unified lifecycle API.
-abstract class LifecycleModule extends SimpleModule with Disposable {
+abstract class _LifecycleModule<T extends _LifecycleModule<T>> with Disposable {
```

Comment:
Note that `SimpleModule` contains `api`, `components`, and `events` - the properties I am attempting to compose out of the new `LifecycleModuleV2`.